hw8/task2.py

```diff
@@ -0,0 +1,68 @@
+import sqlite3
+import os.path
+
+
+class TableData:
+    """
+    This is a wrapper class  for database table.
+    That when initialized with database name and table acts as collection object (implements Collection protocol).
+    """
+
+    def __init__(self, database_name: str, table_name: str):
+        """
+
+        :param database_name: type: str  takes database name
+        :param table_name: type: str takes name of table in database
+        """
+        self.database_name = database_name
+        self.table_name = table_name
+        self.conn = sqlite3.connect(database_name)
+        self.cursor = self.conn.cursor()
+
+    def __len__(self):
+        """
+        This method should return the  current amount of rows in the table, an integer >=0
+        :return: int
+        """
+        self.cursor.execute(f'select count(*) from {self.table_name}')
+        data = self.cursor.fetchall()
+        return data[0][0]
+
+    def __getitem__(self, item):
+        """
+        This method is use to implement call like self[item]
+        :param item: type: str
+        :return: Should return single data row from table with name == 'item'
+        """
+        data = self.cursor.execute(f" SELECT * from {self.table_name}")
+        names = [description[0] for description in data.description]
+        if item in names:
```

Comment:
what this lines are doing names =
 [description[0] for description in data.description]
data = [data[0] for data in self.cursor.execute(f'SELECT {item} from {self.table_name} ').fetchall()]
 return data